content/posts/chaos-testing-guide/index.mdx

```diff
@@ -0,0 +1,228 @@
+
+  - Test the resiliency of a system under turbulent conditions by running tests that are designed to disrupt while monitoring the systems adaptability and performance:
+    - Establish and define your steady state and metrics - understand the behavior and performance under stable conditions  and define the metrics that will be used to evaluate the system’s behavior.  Then decide on acceptable outcomes before injecting chaos. 
+    - Analyze the statuses and metrics of all components during the chaos test runs.
+    - Improve the areas that are not resilient and performant by comparing the key metrics and Service Level Objectives (SLOs) to the stable conditions before the chaos.
+      For example: evaluating the API server latency or application uptime to see if the key performance indicators and service level indicators are still within the acceptable limits.
+
+
+### Best Practices
+Now that we understand the test methodology, let’s take a look at the best practices for an OpenShift cluster.  On that platform there are user applications and cluster workloads that need to be designed for stability and to provide the best user experience possible:
+- Alerts with appropriate severity should get fired
+  - Alerts are key to identify when a component starts degrading,  and can help focus the investigation effort on affected system components.
+  - Alerts should have proper severity, description, notification policy, escalation policy, and SOP in order to reduce MTTR for responding SRE or Ops resources. Detailed information on the alerts consistency can be found [here](https://github.com/openshift/enhancements/blob/master/enhancements/monitoring/alerting-consistency.md).
+
+- Minimal performance impact - Network, CPU, Memory, Storage, Throughput etc. 
+  - The system, as well as the applications, should be designed to have minimal performance impact during disruptions to ensure stability and also to avoid hogging resources that other applications can use.
+  - We want to look at this in terms of CPU, Memory, Storage, Throughput, Network etc.
+
+- Minimal initialization time and fast recovery logic
+  - The controller watching the component should recognize a failure as soon as possible. The component needs to have minimal initialization time to avoid extended downtime or overloading the replicas if it is a highly available configuration. The cause of failure can be because of issues with the infrastructure on top of which it’s running, application failures or because of service failures that it depends on.
+
+- Appropriate CPU/Memory limits set to avoid performance throttling and OOM kills
+  - There might be rogue applications hogging resources ( CPU/Memory ) on the nodes which might lead to applications underperforming  or worse getting OOM killed. It’s important to ensure that the applications and system components have reserved resources for the kube-scheduler to take into consideration in order to keep them performing at the expected levels.
```

Comment:
```suggestion
  - There might be rogue applications hogging resources ( CPU/Memory ) on the nodes which might lead to applications underperforming  or worse getting OOM killed. It’s important to ensure that the applications and system components have reserved resources for the `kube-scheduler` to take into consideration in order to keep them performing at the expected levels.
```

The (proposed) markup style guide uses `backticks` for application or program executable names.

Reply:
Fixed it, thanks.